From patchwork Mon Feb 11 19:51:35 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 8bit
Subject: vc4: Fix copy-and-paste fail in backport of NEON asm fixes.
From: Eric Anholt <eric@anholt.net>
X-Patchwork-Id: 285494
Message-Id: <20190211195135.3005-1-eric@anholt.net>
To: mesa-dev@lists.freedesktop.org
Cc: Emil Velikov <emil.l.velikov@gmail.com>
Date: Mon, 11 Feb 2019 11:51:35 -0800

One of the cpu pointers wasn't marked as read-write, causing gcc to complain:

../src/gallium/drivers/vc4/vc4_tiling_lt.c:181:17: error: output operand constraint lacks ‘=’
                 __asm__ volatile (

Cc: Emil Velikov <emil.l.velikov@gmail.com>
---

This patch is for the 18.3 branch only

 src/gallium/drivers/vc4/vc4_tiling_lt.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/gallium/drivers/vc4/vc4_tiling_lt.c b/src/gallium/drivers/vc4/vc4_tiling_lt.c
index 324a6334668b..167161fdff54 100644
--- a/src/gallium/drivers/vc4/vc4_tiling_lt.c
+++ b/src/gallium/drivers/vc4/vc4_tiling_lt.c
@@ -194,7 +194,7 @@ vc4_store_utile(void *gpu, void *cpu, uint32_t cpu_stride, uint32_t cpp)
                          * d0-d7.
                          */
                         "vstm %[gpu], {q0, q1, q2, q3}\n"
-                        : [cpu]         "r"(cpu)
+                        : [cpu]         "+r"(cpu)
                         : [gpu]         "r"(gpu),
                           [cpu_stride]  "r"(cpu_stride)
                         : "q0", "q1", "q2", "q3");
